concent_api/core/payments/bankster.py

```diff
@@ -264,80 +235,115 @@ def settle_overdue_acceptances(
+            provider_eth_address=provider_ethereum_address,
+            payment_ts=oldest_payments_ts,
+            current_time=current_time,
+            transaction_type=TransactionType.BATCH,
         )
 
-    # Any of the items from list of overdue acceptances
-    # matches condition current_time < payment_ts + PAYMENT_DUE_TIME
-    acceptance_time_overdue = any(
-        current_time < subtask_results_accepted.payment_ts + settings.PAYMENT_DUE_TIME
-        for subtask_results_accepted in acceptances
-    )
+        # Concent defines time T1 equal to youngest timestamp from list of transactions.
+        if len(list_of_transactions) == 0:
+            t1_is_bigger_than_payments_ts = False
+        else:
+            youngest_transaction = max(transaction.closure_time for transaction in list_of_transactions)
+            # Concent checks if all passed SubtaskResultAccepted messages from subtask_results_accepted_list
+            # have payment_ts < T1
+            t1_is_bigger_than_payments_ts = any(
+                youngest_transaction > subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances
+            )
 
-    if t1_is_bigger_than_payments_ts or acceptance_time_overdue:
-        return ClaimPaymentInfo(0, 0, timestamp_error=True)
+        # Any of the items from list of overdue acceptances
+        # matches condition current_time < payment_ts + PAYMENT_DUE_TIME
+        acceptance_time_overdue = any(
+            current_time < subtask_results_accepted.payment_ts + settings.PAYMENT_DUE_TIME
+            for subtask_results_accepted in acceptances
+        )
 
-    # Concent gets list of forced payments from payment API where T0 <= payment_ts + PAYMENT_DUE_TIME.
-    list_of_forced_payments = service.get_list_of_payments(  # pylint: disable=no-value-for-parameter
-        requestor_eth_address=requestor_ethereum_address,
-        provider_eth_address=provider_ethereum_address,
-        payment_ts=oldest_payments_ts + settings.PAYMENT_DUE_TIME,  # Im not sure, check it please
-        current_time=current_time,
-        transaction_type=TransactionType.FORCE,
-    )
+        if t1_is_bigger_than_payments_ts or acceptance_time_overdue:
+            return 'timestamp_error'
 
-    (_amount_paid, amount_pending) = get_provider_payment_info(
-        list_of_forced_payments=list_of_forced_payments,
-        list_of_payments=list_of_transactions,
-        subtask_results_accepted_list=acceptances,
-    )
+        # Concent gets list of forced payments from payment API where T0 <= payment_ts + PAYMENT_DUE_TIME.
+        list_of_forced_payments = service.get_list_of_payments(  # pylint: disable=no-value-for-parameter
+            requestor_eth_address=requestor_ethereum_address,
+            provider_eth_address=provider_ethereum_address,
+            payment_ts=oldest_payments_ts + settings.PAYMENT_DUE_TIME,  # Im not sure, check it please
+            current_time=current_time,
```

Comment:
It might have been based on the fact that we don't don't let provider start the procedure earlier than `payments_ts + settings.PAYMENT_DUE_TIME`. But it's not correct as T0. Payment after T0 is fine. But a payment after T0 + `settings.PAYMENT_DUE_TIME` is late. Using this interval would make Concent not see any payment that was made on time.

Reply:
Looks OK now. I think you should remove the `# Im not sure, check it please` comment now.

Reply:
:+1: done